server.ts

```diff
@@ -33,7 +35,6 @@ global['navigator'] = win.navigator;
 export function app() {
   const logging = /on|1|true|yes/.test(process.env.LOGGING?.toLowerCase());
 
-  const ICM_BASE_URL = process.env.ICM_BASE_URL || environment.icmBaseURL;
```

Comment:
Moving this to a larger scope means that it lacks a "falsey" check at the right time. You are accessing this value in function `run()` but the falsey check is performed in `app` function. To late...